### DISCUSSION TITLE
JVM target 17

### DISCUSSION THREAD

**Discussion by vladimirfx** [detekt/detekt#4224] [Q&A] [answered]
Kotlin 1.6-RC allows JVM target 17. How to support this target in Detekt type resolution tasks?

**Comment by cortinico** [comment, marked as answer]
> Kotlin 1.6-RC allows JVM target 17. How to support this target in Detekt type resolution tasks?

We haven't looked into 1.6.x yet. Have you tried to specify `"17"` for `jvmTarget`? Is it failing with any error message?

We're currently also waiting for Gradle 7.3 which will support building projects with Java 17: https://docs.gradle.org/7.3-rc-1/release-notes.html so I'd say that you're a bit earlier to the game :)

**Reply by vladimirfx** [reply]
Yes, I try to form the 'next year' stack. Now it is:
- JDK 17
- Gradle 7.3 RC3
- Kotlin 1.6 RC
- Detect 1.19 RC1

It is very likely that the next stable versions of Gradle, Kotlin, and Detekt are released at roughly the same time and only Detekt (in type resolution mode) do not support JVM 17. It is deferring the adoption of JDK 17 in Kotlin projects that uses Detekt. 

The error message is exactly as for Kotlin 1.5:
```log
Execution failed for task ':detektMain'.
> Invalid value passed to --jvm-target

```
So I've tried to understand how to switch to Kotlin 1.6 in Detekt dependency graph and should it work at all.

**Reply by vladimirfx** [reply]
I've successfully overrides embedded Kotlin compiler and it sims work:

```Kotlin
dependencies {
    detekt("io.gitlab.arturbosch.detekt:detekt-cli:1.19.0-RC1")
    detekt("org.jetbrains.kotlin:kotlin-compiler-embeddable:1.6.0-RC")
    detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:1.19.0-RC1")
}
```
